### DISCUSSION TITLE
Auth::user()->id problem

### DISCUSSION THREAD

**Discussion by fatihbarissahan** [spatie/laravel-permission#2123] [Q&A] [answered]
Hello,

I tried every way but I can't figure it out. I am using Yajra DataTable plugin and I need to hide some fields based on login user's permissions. Whatever I did, I get an error when I want to get the id of the logged in user. It works when I give the id information myself (example: user with id 1). How can I solve this problem?

![screen-shot-1-1](https://user-images.githubusercontent.com/65877282/175832701-5cfa5647-d28e-4b18-b5f7-e4c21a26f341.png)
![screen-shot-1](https://user-images.githubusercontent.com/65877282/175832703-0a31db6b-87cb-4d11-9ceb-1a1404addf34.png)

![screen-shot-2](https://user-images.githubusercontent.com/65877282/175832641-3b60f204-25b2-4af2-82e1-7ca8d4e4d709.png)
![screen-shot-3](https://user-images.githubusercontent.com/65877282/175832642-8c6a5919-4b2f-461c-90b0-aff3bfeaa728.png)

Another way:

![Screen Shot 2022-06-27 at 12 57 26](https://user-images.githubusercontent.com/65877282/175915628-a516ea47-467d-4153-b764-6fcfc88ff70e.png)
![Screen Shot 2022-06-27 at 12 58 10](https://user-images.githubusercontent.com/65877282/175915675-2b03b053-9ee6-451f-8092-22c369ccaf04.png)
![Screen Shot 2022-06-27 at 12 58 32](https://user-images.githubusercontent.com/65877282/175915749-be19feec-150f-4ba2-8cb2-69ca97b801a3.png)

Here 

User::with('roles')->where('id',**Auth::id()**)->first(); in its place
User::with('roles')->where('id',**1**)->first(); It works normally without any problems.

**Comment by erikn69** [comment, marked as answer]
Make a debug

**Reply by fatihbarissahan** [reply]
I've done everything. Normally, I get the id of the logged in user, no problem. But when I try to use it somewhere in the datatable, I cannot access the information of the logged in user.

**Reply by erikn69** [reply]
Maybe it uses ajax and you are not sending csrf or bearer token

**Reply by fatihbarissahan** [reply]
I solved this problem.

![Screen Shot 2022-06-27 at 22 50 49](https://user-images.githubusercontent.com/65877282/176024209-c6001f7c-5eab-40bb-8c2f-de21e971bc6b.png)
![Screen Shot 2022-06-27 at 22 51 19](https://user-images.githubusercontent.com/65877282/176024214-f3fc3f46-c23a-48e8-bb03-8cad34172f90.png)
